tesla_api/energy.py

```diff
@@ -50,10 +56,33 @@ async def get_battery_count(self):
         info = await self.get_energy_site_info()
         return int(info["battery_count"])
 
-
     async def get_energy_site_live_status(self):
         return await self._api_client.get('energy_sites/{}/live_status'.format(self._energy_site_id))
 
+    async def get_energy_site_calendar_history_data(self, kind: str = 'energy', period: str = 'day',
+                                                    end_date: [str, datetime] = None) -> Optional[dict]:
+        """
+        Returns historical energy data
+        :param kind: [power, energy, self_consumption]
+        :param period: Amount of time to include in report. One of day, week, month, year, and lifetime.
+                    When kind is 'power', this parameter is ignored, and the period is always 'day'
+        :param end_date: ISO 8601 datetime, e.g. 2019-12-23T17:39:18.546Z. either in str or as timezone aware datetime.
+                    The response report interval ends on this datetime and starts at the beginning of the given period.
+                    Note time cannot be 00:00, so if the datetime object has no hour and minute, it will be set to 23:59. 
+                    This allows you to call with datetime(year=2020, month=5, day=1) and get all data for May 1st.
+                    Defaults to the current time.
+        """
+        params = {'kind': kind, 'period': period}
+        if end_date is not None:
+            if isinstance(end_date, datetime):
+                if end_date.hour == 0 and end_date.minute == 0:
+                    # if the datetime object's time is 00:00 then the API returns nothing. We adjust by adding 23:59
+                    # so you can use datetime(year=2020, month=5, day=2) and it gets the data for May 2, 2020 as expected
+                    end_date = end_date + timedelta(hours=23, minutes=59)
+            params['end_date'] = end_date.isoformat() if isinstance(end_date, datetime) else end_date
```

Comment:
Makes sense to support date directly then:
```suggestion
        if isinstance(end_date, date):
            if not isinstance(end_date, datetime):
                end_date = datetime.combine(end_date, time(23, 59, 59))
            elif end_date.hour == 0 and end_date.minute == 0:
                # If the datetime object's time is 00:00 then the API returns nothing. We adjust by adding 23:59,
                # so you can use datetime(year=2020, month=5, day=2) and it gets the data for May 2, 2020 as expected
                end_date = end_date.replace(hour=23, minute=59, second=59)
            end_date = end_date.isoformat()
        params['end_date'] = end_date
```